### ISSUE TITLE
can't download form download.bell-sw.com

### ISSUE THREAD

**Issue by mosinnik** [paketo-buildpacks/bellsoft-liberica#488]
Tring to build native image for opentelemetry example (https://github.com/open-telemetry/opentelemetry-java-examples/blob/main/spring-native/README.md) faced with error while loading base image:
```
...
    [creator]         Using Java version 17 extracted from MANIFEST.MF
    [creator]       BellSoft Liberica NIK 17.0.7: Contributing to layer
    [creator]         Downloading from https://download.bell-sw.com/vm/23.0.0/bellsoft-liberica-vm-core-openjdk17.0.7+7-23.0.0+1-linux-amd64.tar.gz
    [creator]     unable to invoke layer creator                                                                                                 
    [creator]     unable to get dependency native-image-svm                                                                                      
    [creator]     unable to download https://download.bell-sw.com/vm/23.0.0/bellsoft-liberica-vm-core-openjdk17.0.7+7-23.0.0+1-linux-amd64.tar.gz
    [creator]     could not download https://download.bell-sw.com/vm/23.0.0/bellsoft-liberica-vm-core-openjdk17.0.7+7-23.0.0+1-linux-amd64.tar.gz: 403
    [creator]     ERROR: failed to build: exit status 1
```
Trying to open in brawser found that https://download.bell-sw.com/ has guarded by cloudflare and it is blocked and as usually no way to reset restriction or understand cause.

So need wide open resource to download from or remove cloudflare guard for `https://download.bell-sw.com/vm/ `.

**Comment by dmikusa** [comment]
We do not control the download locations and do not have the resources to rehost the downloads used by buildpacks.

Options you have:

1. Push your traffic through a proxy or VPN so that it can side-step whatever CloudFlare is doing or complaining about. See https://paketo.io/docs/howto/configuration/#proxy-configuration
2. Download the Bellsoft Liberica in your browser or through some other means, like directly from their website (the one we use is the Linux "core" package. Then you can use a dependency mapping to point the buildpack to a different location from which it can fetch the dependency (like on a server you control, or as a file you volume map into the build container). https://paketo.io/docs/howto/configuration/#dependency-mappings. See also https://github.com/dmikusa/binding-tool/ which provides some help doing this.
3. Use GraalVM instead. See https://paketo.io/docs/howto/java/#use-an-alternative-java-native-image-toolkit. This will simply use a different download. This one comes from Github.

Hope that helps!

**Comment by mosinnik** [comment]
so problem should be fixed by open-telemetry and they must change graavm builder algorithm?

**Comment by dmikusa** [comment]
No, I'd imagine the demo you referenced is fine. It your network. You're getting a 403 when you try to access that resource, that's an issue specific to your network. The same requests work just fine for me.

If you're on a corporate network, the first thing I'd try is 1.) from above. Many corp networks require a proxy and you could get a 403 if you're not going through the proxy. Besides that, I can't really say. Networks vary a lot so you could be hitting something more complicated.

The other options 2.) and 3.) above are just other ways you could attempt to get around your network issue, if you can't resolve the network issue.

**Comment by mosinnik** [comment]
I think that I know what 403 error about - just bellsoft admins setup cloudflare with some strange restrictions rules that applied for something that nowhere described and as I see have no expiration time. 
All variant are bad because of expected that example project that using paketo will be built from scratch without any unnecessary movements. And I assume there are also people who may faced same issue because of partially links are refer to github (like JRE) with no restrictions and other part (like NIK) that may be restricted. 
So as a solution I see that paketo should refer to unrestricted resources only. Of course it may be not trivial to check if resource is restricted for each links, but in this case I already notified about this.

I don't know who update versions and links in this bellsoft repo but if those people have relations with bellsoft repo owners  must make then to give links that are not restricted and even better to share NIK images on github like JRE

**Comment by dmikusa** [comment]
I don't know of NIK issues on Github. We use the JRE/JDK links on Github because they are there. When I was looking around this morning I didn't see them. We also get complaints about download speeds from Github in a few regions of the world. I believe that's not the same with the downloads from Bellsoft, so there's always something.

I can't promise anything will change, but we'll pass this feedback along to our contact at Bellsoft. Thanks for the report and details.

**Comment by anthonydahanne** [comment]
Answer from Bellsoft:
>   [Bell-sw.com](http://bell-sw.com/) is using CDN from CloudFlare, which has both region (based on risk mitigation) and AI-based rules. To clarify the root cause of the problem, it’s suggested to write to [info@bell-sw.com](mailto:info@bell-sw.com) with a Ray ID from CF.
> That said, based on the bandwidth we start to get with NIK, it may be time we move the hosting of NIK binaries in REST API response to [github.com](http://github.com/), like Liberica.

As a matter of fact, only NIK downloads come from download.bell-sw.com

**Comment by sergey-morenets** [comment]
Hi

I have similar issue but I can access bell-sw.com from my browser.

**Comment by dmikusa** [comment]
Ok, if anyone that has posted here or comes across this in the future has issues with NIK downloads from bell-sw.com, please reach out using the information provided above: https://github.com/paketo-buildpacks/bellsoft-liberica/issues/488#issuecomment-1788087066
